### ISSUE TITLE
Monorepo/Yarn Workspaces Compatibility

### ISSUE THREAD

**Issue by phobetron** [redaxmedia/handpick#20]
Does `handpick` support monorepos? I've attempted to use it within a monorepo package, but I receive the following error:

```plain text
TypeError: Cannot convert undefined or null to object
```

Specifically, I'm using the latest version of Yarn.

**Comment by redaxmedia** [comment]
Hey,

it's hard to figure out what goes wrong here. Is the `package.json` valid and working with yarn? The best would be to share your repository with me if possible.

**Comment by phobetron** [comment]
@redaxmedia Sorry for the delayed response. I think it's reasonable to expect one workspace argument if workspaces are configured. That would be pointing in the same direction as Yarn's behavior, at least.

**Comment by redaxmedia** [comment]
Good, I try to make this feature available in the next release...

**Comment by redaxmedia** [comment]
Update: The latest prerelease for 4.0.0 allow the usage of workspaces but I would not call this fully supported.

Handpick works like this:

1. Manipulate `package.json`by merging everything into a temporary `dependencies`property
1. Run the package manager in a child process
1. Revert `package.json`to it's original stage once the child process ended or was terminated

To have full workspace support, the following features are needed:

1. Pass arguments that are scoped by the workspace like `a.target=lintDependencies b.filter=devDependencies`
1. Manipulate at least 3 x `package.json` files (root, workspace a, workspace b)
1. Interoperate between all workspaces in terms of their targeted, filtered and referenced dependencies

In other words: it is complicated and I suggest to move shared dependencies to the root `package.json` and work from there. At this point I have no idea for a good API that is worth to be implemented.

**Comment by redaxmedia** [comment]
I close the issue for the moment, just in case you have a proper solution: let me know
